### PR TITLE
Windows friendly paths

### PR DESCRIPTION
**Pull request by ocefpaf**
@efiring this is an effort to make it easier and more transparent the steps to reproduce the data and code here.

Also, if you agree, I would like to vendorize `loadmatbunch` to avoid the dependency on `pycurrents` b/c it is not easily available on `PyPI` or `conda`.

### CODE REVIEW

**Comment by efiring** [comment]
OK, you can go ahead with this and with including loadmatbunch--or for purposes here, can (and should) it be replaced with a few lines using scipy.io.loadmat directly?

File: tools/mat2npz.py [hunk 1]
```diff
@@ -4,19 +4,47 @@
...
+
+
+def loadmatdict(fname):
+    out = {}
+    with fname.open('rb') as fobj:
+        xx = sio.loadmat(fobj)
+        keys = [k for k in xx.keys() if not k.startswith('__')]
+        for k in keys:
+            out[k] = _structured_to_dict(xx[k])
+    return out
```
Comment (ocefpaf) [inline tools/mat2npz.py]: @efiring is this OK?

**Comment by efiring** [comment]
Pathlib support in the standard library is spotty...
```
(test) ~/work/programs/TEOS-10/github/GSW-Python/tools $ python mat2npz.py
Traceback (most recent call last):
  File "mat2npz.py", line 50, in <module>
    np.savez(fname, **cv_vars)
  File "/Users/efiring/anaconda/envs/test/lib/python3.5/site-packages/numpy/lib/npyio.py", line 574, in savez
    _savez(file, args, kwds, False)
  File "/Users/efiring/anaconda/envs/test/lib/python3.5/site-packages/numpy/lib/npyio.py", line 642, in _savez
    zipf.write(tmpfile, arcname=fname)
  File "/Users/efiring/anaconda/envs/test/lib/python3.5/zipfile.py", line 1488, in write
    self.fp.write(zinfo.FileHeader(zip64))
  File "/Users/efiring/anaconda/envs/test/lib/python3.5/zipfile.py", line 680, in write
    n = self.fp.write(data)
AttributeError: 'PosixPath' object has no attribute 'write'
Exception ignored in: <bound method ZipFile.__del__ of <zipfile.ZipFile [closed]>>
Traceback (most recent call last):
  File "/Users/efiring/anaconda/envs/test/lib/python3.5/zipfile.py", line 1595, in __del__
    self.close()
  File "/Users/efiring/anaconda/envs/test/lib/python3.5/zipfile.py", line 1608, in close
    self._write_end_record()
  File "/Users/efiring/anaconda/envs/test/lib/python3.5/zipfile.py", line 1711, in _write_end_record
    self.fp.write(endrec)
  File "/Users/efiring/anaconda/envs/test/lib/python3.5/zipfile.py", line 680, in write
    n = self.fp.write(data)
AttributeError: 'PosixPath' object has no attribute 'write'
```

**Comment by efiring** [comment]
The traceback above is with python 3.5; I haven't tried 3.6.

**Comment by efiring** [comment]
It's probably the numpy version, then.  I'm on 1.11.3.

**Comment by ocefpaf** [comment]
> It's probably the `numpy` version, then. I'm on 1.11.3.

Indeed. Supporting that version is relatively easy (see my last commit).

**Pull request merged**